### PR TITLE
Distribution View Updates (previously: A few boolean options (previously: Stacked bar bool))

### PR DESCRIPTION
**Pull request by chutten**
While prototyping how bool probes might look I stumbled upon a fifth option: stacking into a single plot that sums to 100%.

I think it would display better horizontally, but `orientation: "h"`isn't working immediately (maybe it's because of my `yaxis/xaxis` `layout` customizations? Something to investigate.)

Live demo's not working so well due to a weird bug that's messing with the live-hosted portals

### CODE REVIEW

**Comment by georgf** [comment]
I assume we'll also still add the other proposed boolean graphing options so we can get feedback on them?

File: src/components/distributionview.jsx [hunk 1]
```diff
@@ -40,32 +69,36 @@ export class DistributionView extends Component {
...
-      name: metric,
-      type: "bar",
-      x: [],
-      y: [],
-      text: [],
-      customdata: [],
-      hoverinfo: "y+text",
-    };
+    const BOOL_MEASURES = [
+      'scalars_devtools_onboarding_is_devtools_user',
```
Comment (georgf) [inline src/components/distributionview.jsx]: Note for future work: Later we might want to move this into `MetricData`, so we can share the type information across different views.

**Comment by chutten** [comment]
That's what I'm working on now, yes. I'm also generating a little more boolean data so we don't have to judge based on only one shape of data.

File: src/components/distributionview.jsx [hunk 2]
```diff
@@ -40,32 +69,36 @@ export class DistributionView extends Component {
...
-        x = start;
-        text = `[${start}, ${end-1}) - ${this.formatCount(count)} clients`;
-      }
-      plotlyData.x.push(x);
-      plotlyData.text.push(text);
-      plotlyData.y.push(proportion);
-      plotlyData.customdata.push(count);
+    let plotData;
+    if (BOOL_MEASURES.includes(metric)) {
+      // need to calculate "sometimes true" and "sometimes false" data
```
Comment (georgf) [inline src/components/distributionview.jsx]: Style/readability: Can we move the code in this branch to a helper function?

**Review by georgf — APPROVED** [review]
This looks good to me as an experiment, assuming we're also adding the other options for exploration and feedback.

It might be worth it to move the boolean distributions to a different (sub) component, so we're more free to experiment with them without concerns for the other distributions.

**Comment by chutten** [comment]
Here are the multiple boolean plots and some additional data. Next I'll work on encapsulation a bit.

File: src/components/distributionview.jsx [hunk 3]
```diff
@@ -64,6 +64,59 @@ export class DistributionView extends Component {
     window.removeEventListener("resize", this.onResize);
   }
 
+  makeBooleanData = (metric, [never, always]) => {
```
Comment (georgf) [inline src/components/distributionview.jsx]: Style: No need for a lambda, just make this `makeBooleanData(...) { ...`, like `render()` below.

**Comment by chutten** [comment]
Live demo should update shortly. This is back up for review.

File: src/components/distributionview.jsx [hunk 4]
```diff
@@ -73,6 +113,7 @@ export class DistributionView extends Component {
             <i className="fas fa-info-circle"></i> The distribution view displays the distribution of user outcomes as a histogram.
             <div>Hover over a point on the graph to view a specific value.</div>
             <div>Or, switch to table mode below to see a list of all absolute values.</div>
+            {extraMessage}
```
Comment (georgf) [inline src/components/distributionview.jsx]: This looks disconnected to me from the rest of the presentation, it requires me to look at the text for one kind of statement and at the graph for other statements. I think we should:
- not present this like a caption and
- not make this the only statement that's presented in this form

I'd suggest to instead:
- Move the statement down below the graph. (makes it more clearly supplementary) 
- Not make it `<h4>`, just plain text, and iterate on design from there in a separate conversation.
- Add a statement for each of the important things we want people to read out of the data (always true, always false, at least once true, at least once false ?).

I'm happy to see the additional statements added in a follow-up if that helps.

File: src/components/distributionview.jsx [hunk 5]
```diff
@@ -35,35 +64,46 @@ export class DistributionView extends Component {
...
-      plotlyData.y.push(proportion);
-      plotlyData.customdata.push(count);
+    let plotData;
+    let extraMessage;
+    if (BOOL_MEASURES.includes(metric)) {
+      plotData = this.makeBooleanData(metric, data);
+      const trueCount = Math.ceil(data[0].count / data[0].proportion) - data[0].count;
+      const trueProportion = ((1 - data[0].proportion) * 100).toFixed(2);
+      extraMessage = (
+        <h4>Clients reporting true at least once for {metric}: {trueCount} ({trueProportion}%)</h4>
```
Comment (georgf) [inline src/components/distributionview.jsx]: - Below we use the "users" terminology, lets stay consistent with that.
- Wording could be more natural, e.g. "X% of users (N) are reporting `true` at least once.". (happy to iterate on this part in a follow-up)

**Review by georgf — APPROVED** [review]
Cheers, this looks good overall given yesterdays feedback.
This is good to merge when the issues commented on are resolved (or a followup issue filed for the once marked as potential followups).

A data issue:
Switching to scalars_devtools_onboarding_is_devtools_user for nightly 62, looking at the tooltip, the graph has "always" rounded to 100%, which mismatches the other values around.

A small design nit for a follow-up:
When i hover over one bar, i get tooltips for all three bars attached to the bar i'm over. This seems a bit confusing?
We can either always show all, but attached to their bar, or only show one at a time.

**Comment by chutten** [comment]
I mentioned it in the commit, and it'll come up again in #62, but in rewording `extraText` I had to write it in terms of counts first and proportions parenthetically because English. We can spitball better texts in #62, as I understand we want to this dashboard to be Proportions First (raw counts second).

**Pull request merged**